### PR TITLE
Feature/shared refactor

### PR DESCRIPTION
**Pull request by waterplea**

### CODE REVIEW

**Comment by dr-bonez** [comment]
@waterplea `git rebase HEAD~3 --onto origin/feature/marketplace-refactor`
I can't push to your branch, so I can't do it for you.

File: frontend/package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "patch-db-client": "file: ../../../patch-db/client",
     "pbkdf2": "^3.1.2",
     "rxjs": "^6.6.7",
+    "src": "^1.1.2",
```
Comment (MattDHill) [inline frontend/package.json:52]: Where is this being used?

File: frontend/projects/shared/src/components/status/status.component.module.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core'
```
Comment (MattDHill) [inline frontend/projects/shared/src/components/status/status.component.module.ts:1]: status component is only used in the ui project.

File: frontend/projects/shared/src/const/primary-rendering.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { PrimaryStatus, StatusRendering } from '../types/status'
```
Comment (MattDHill) [inline frontend/projects/shared/src/const/primary-rendering.ts:1]: This is also only for UI project.

File: frontend/projects/shared/src/types/config-types.ts [hunk 4]
```diff
@@ -1,6 +1,4 @@
-export interface ConfigSpec {
-  [key: string]: ValueSpec
-}
+export type ConfigSpec = Record<string, ValueSpec>
```
Comment (MattDHill) [inline frontend/projects/shared/src/types/config-types.ts:1]: config-types are only UI project.

File: frontend/projects/shared/src/types/status.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+export enum PrimaryStatus {
```
Comment (MattDHill) [inline frontend/projects/shared/src/types/status.ts:1]: only UI project

File: frontend/projects/ui/src/app/pages/marketplace-routes/marketplace-show/marketplace-show.module.ts [hunk 6]
```diff
@@ -3,9 +3,15 @@ import { CommonModule } from '@angular/common'
...
 import { IonicModule } from '@ionic/angular'
 import { MarketplaceShowPage } from './marketplace-show.page'
-import { SharingModule } from 'src/app/modules/sharing.module'
-import { StatusComponentModule } from 'src/app/components/status/status.component.module'
+import {
+  SharedPipesModule,
+  EmverPipesModule,
+  MarkdownPipeModule,
+  TextSpinnerComponentModule,
+  StatusComponentModule,
```
Comment (MattDHill) [inline frontend/projects/ui/src/app/pages/marketplace-routes/marketplace-show/marketplace-show.module.ts:11]: We're not actually using this, can delete and remove status component from shared lib.

File: frontend/package.json (hunk 1 above)
Comment (waterplea) [reply frontend/package.json:52]: I didn't add anything myself. @dr-bonez could it be something with rebase? Maybe it was there when I started but then got removed but rebase did not sync it? I'll remove it.

**Review by MattDHill — APPROVED** [review]

**Pull request merged**